dev/TeachingTip/TeachingTip.xaml

```diff
@@ -550,22 +550,30 @@
                                             Child="{TemplateBinding HeroContent}"
                                             Background="{TemplateBinding Background}"/>
                                     <Grid x:Name="NonHeroContentRootGrid" Grid.Row="1">
-                                        <ScrollViewer Foreground="{TemplateBinding Foreground}"
-                                                      VerticalScrollBarVisibility="Auto">
+                                        <ScrollViewer VerticalScrollBarVisibility="Auto">
                                             <StackPanel Margin="{StaticResource TeachingTipContentMargin}">
                                                 <Grid Grid.Row="0">
                                                     <Grid.ColumnDefinitions>
                                                         <ColumnDefinition Width="Auto"/>
                                                         <ColumnDefinition Width="*"/>
                                                     </Grid.ColumnDefinitions>
 
-                                                    <Border x:Name="IconBorder" Grid.Column="0" Child="{Binding RelativeSource={RelativeSource TemplatedParent}, Path=TemplateSettings.IconElement}"/>
+                                                    <ContentPresenter x:Name="IconPresenter"  Grid.Column="0" Foreground="{TemplateBinding Foreground}"
```

Comment:
>ContentPresenter [](start = 53, length = 16)

We had an issue when this was a ContentPresenter in the past.  Content presenter does some magic so that you don't have to set its content manually, if it is unset, it looks at the meta data of the controls properties for the default property and sets that as the ContentPresenter's content.  What this means is that if you have a teaching tip which does not have an icon but does have something for its content property, this will result in that content property being multiparented and crashing.  I think this should stay a border.

Reply:
I think #349 was the original issue but the details look lacking.

Reply:
What element should we parent the icon in then, if we want the foreground to also affect it? Border does not have a foreground property, similarly to other lightweight controls. I switched to ContentPresenter since that has a Foreground property, however if that is problematic, what should we do here?

Reply:
ah hmm... kind of annoying. perhaps it should be a content presenter whose child is a border that we put the icon in? that way we can have a target for the foreground forwarding and avoid the content presenter issue by guaranteeing a child other than the default property?

Reply:
That's an excellent idea! While this is a bit more expensive on the visual tree then before hand, I think it's fine. TeachingTip isn't one of those controls that appear hundreds of times on the screen so that should be fine. So to summarize, current state of the PR:

Properties get passed down to:
- Icon
- Content

Following parts are unaffected:
- HeroContent
- CloseButton
- ActionButton

Title and Subtitle have lightweightstyling resources for their foreground, fontweight etc are not customizable.

Reply:
I think that makes sense because the icon is independently customization by setting the foreground on the icon source (once the bug fix goes in)

Reply:
Is there anything else needed to do in this PR then or can we run the CI and merge after that?